bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdPreferredOptions.java

```diff
@@ -19,21 +19,25 @@
 import java.util.Objects;
 import java.util.Optional;
 
+import org.eclipse.cdt.lsp.clangd.ClangdEnable;
 import org.eclipse.cdt.lsp.clangd.ClangdMetadata;
 import org.eclipse.cdt.lsp.clangd.ClangdOptions;
+import org.eclipse.core.resources.IProject;
 import org.eclipse.core.runtime.preferences.IScopeContext;
 import org.eclipse.core.runtime.preferences.PreferenceMetadata;
 
-final class ClangdPreferredOptions implements ClangdOptions {
+final class ClangdPreferredOptions implements ClangdOptions, ClangdEnable {
 
 	private final String qualifier;
 	private final IScopeContext[] scopes;
 	private final ClangdMetadata metadata;
+	private final ClangdEnable enable;
```

Comment:
It would be better to avoid `null` field but we may rework it later

Reply:
That's right. I made this field optional, since it should only used when there is a service provider available. I'll open a new issue, to improve this handling in the future.

Reply:
Would it be better if the `ClangdEnable` interface method `isEnabledFor` would return  an`Optional<Boolean>`:
```java
public Optional<Boolean> isEnabledFor(IProject project);
```
and the `org.eclipse.cdt.lsp.clangd` bundle would provide a `DefaultClangdEnable` service which returns an empty Optional?

Reply:
current
```Java
public boolean isEnabledFor(IProject project)
```
is good enough. We can think about better impl later.